src/Symfony/Component/DependencyInjection/Compiler/ResolveParameterPlaceHoldersPass.php

```diff
@@ -45,7 +46,7 @@ public function process(ContainerBuilder $container)
             $aliases = array();
             foreach ($container->getAliases() as $name => $target) {
                 $this->currentId = $name;
-                $aliases[$this->bag->resolveValue($name)] = $this->bag->resolveValue($target);
+                $aliases[$this->bag->resolveValue($name)] = (new Alias($this->bag->resolveValue((string) $target)))->setPublic($target->isPublic())->setPrivate($target->isPrivate());
```

Comment:
was the old code actually resolving any parameter in aliases ? AFAIK, `ParameterBag::resolveValue($object)` is a no-op

Reply:
was a no-op yes, so that's a bug fix (but not one we have to do on lower branches, can be seen as a new feat now, isn't it?)

Reply:
well, I'm not sure we actually want to introduce this new feature here (especially not hidden inside a commit saying it turns aliases to private by default).
This should be discussed separately.

Reply:
Btw, I'm sure we had several discussions in the past *rejecting* the addition of this feature. I don't remember whether another discussion after that decided to add it (and the implementation failed at it)

Reply:
Looking at the history, this non-working code was added before the 2.0.0 release. So this is clearly *before* discussions where we rejected this feature.
So you should not add a rejected feature here. I would rather remove this no-op code from the codebase

Reply:
well, the key is still resolved, what should we do about it?

Reply:
Line reverted here as it's unrelated, I'll open another PR on 2.7 about this.